io.fairyproject.modules/platform.mc/module.protocol/mc-protocol/src/main/java/io/fairyproject/mc/protocol/PacketListenerService.java

```diff
@@ -0,0 +1,104 @@
+package io.fairyproject.mc.protocol;
+
+import io.fairyproject.container.ComponentHolder;
+import io.fairyproject.container.ComponentRegistry;
+import io.fairyproject.container.PreInitialize;
+import io.fairyproject.container.Service;
+import io.fairyproject.mc.protocol.annotation.Listen;
+import io.fairyproject.mc.protocol.listener.PacketListener;
+import io.fairyproject.mc.protocol.packet.Packet;
+import io.fairyproject.reflect.Reflect;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.*;
+
+@Service
+public final class PacketListenerService {
+    private final Map<PacketListener, List<InternalListener>> objectListenerCache = new HashMap<>();
+    private final Queue<InternalListener> listeners = new PriorityQueue<>((a, b) -> Integer.compare(b.getPriority(), a.getPriority()));
+
+    @PreInitialize
+    public void preInit() {
+        ComponentRegistry.registerComponentHolder(ComponentHolder.builder()
+                .type(PacketListener.class)
```

Comment:
PacketListener sounds cool, but what about using the built-in event bus system to deploy packet events?
Simply create an event class, annotated as `@Event` and use EventBus.call() to deploy the event.
Check here:
https://github.com/FairyProject/fairy/tree/v5/io.fairyproject.platforms/core-platform/src/main/java/io/fairyproject/event
https://github.com/FairyProject/fairy/blob/v5/io.fairyproject.platforms/mc-platform/src/main/java/io/fairyproject/mc/event/MCPlayerJoinEvent.java

Reply:
We'd be missing out on the pre-done wrappers being injected as a direct parameter, which honestly would save a lot of time from doing wrapper.getPacket() then casting etc...

Reply:
I see, then it should be cool to have both I guess?